core/src/main/java/de/danoeh/antennapod/core/feed/FeedMedia.java

```diff
@@ -369,34 +315,34 @@ public void writeToPreferences(Editor prefEditor) {
 
     @Override
     public void loadMetadata() throws PlayableException {
-        if (item == null && itemID != 0) {
-            item = DBReader.getFeedItem(itemID);
-        }
+        //nothing to process, metadata loaded from database in PlayableUtils
     }
 
     @Override
     public void loadChapterMarks() {
-        if (item == null && itemID != 0) {
-            item = DBReader.getFeedItem(itemID);
-        }
-        if (item == null || item.getChapters() != null) {
-            return;
-        }
-        // check if chapters are stored in db and not loaded yet.
-        if (item.hasChapters()) {
-            DBReader.loadChaptersOfFeedItem(item);
+        if(localFileAvailable()) {
+            ChapterUtils.loadChaptersFromFileUrl(this);
```

Comment:
I think the only place where the `loadChapterMarks` method is used is the chapters screen. Instead, I would just move the loading code there and remove the method here.

Reply:
It shows me that its loaded in two places, in `ChaptersFragment` `private void loadMediaInfo()` and `PlaybackServiceTaskManager` `public synchronized void startChapterLoader(@NonNull final Playable media)`.
I didn't remove loading from DB completely. It is still loaded from DB but in dedicated class `PlayableUtil`.

Reply:
How about creating a method `ChapterUtils.loadChapterMarks` that handles all in one place instead of doing some parts here and some parts in `PlayableUtil`?